src/tools/pub/automated-publishing.md

```diff
@@ -0,0 +1,719 @@
+modifying the workflow file to circumvent
+[environment protection rules][7].
+
+In your GitHub repository settings, you can use
+[environment protection rules][7] to set _required reviewers_.
+If you configure this option, GitHub prevents actions with
+the environment from running until one of the required reviewers have
+approved the run.
+
+![GitHub Action waiting for deployment review](gh-pending-review.png)
+
+
+## Publishing from Google Cloud Build
+
+To publish from [Google Cloud Build][9] you must:
+1. Register a Google Cloud Project.
+1. Create a [service account][10] for publishing to `pub.dev`.
+1. Enable automated publishing for the package on `pub.dev`.
+   Add the email of the service account you created.
+1. Grant the default Cloud Build service account permission to impersonate the
+   service account created for publishing.
+1. Create a `cloudbuild.yaml` file which obtains a temporary OIDC `id_token`
+   and uses it for publishing to `pub.dev`.
+1. Configure a Cloud Build trigger, for running the steps in `cloudbuild.yaml`
+   in your project on Google Cloud Build.
+
+
+{{site.alert.note}}
+  When you enable automated publishing from a _service account_ you must carefully
+  review who has the ability to impersonate this service account, either by
+  calling through APIs, exporting service account keys or through changing
+  IAM permission in the cloud project.
+  To learn more, check out [managing service account impersonation][11].
+{{site.alert.end}}
+
+
+### Create a service account for publishing
+
+To create a service account that can publish to `pub.dev`, you must:
+
+1. Create a cloud project, if you don't have an existing project
+   check out [creating and managing projects][12]. 
+1. Create a _service account_. This account needs permission to publish your
+   package on `pub.dev` and must exist in the same cloud project where you run
+   Cloud Build.
+   Run the following command to create a service account:
+
+   ```console
+   gcloud iam service-accounts create pub-dev \
+     --description='Service account to be impersonated when publishing to pub.dev' \
+     --display-name='pub-dev'
+   ```
+
+   This command results in a service account named:
+   `pub-dev@$PROJECT_ID.iam.gserviceaccount.com`.
+
+1. Grant the service account permission to publish your package.
+
+   This step requires _uploader_ permission on the package or
+   _admin_ of the publisher that owns the package.
+
+   1. Navigate to the **Admin** tab (`pub.dev/packages/<package>/admin`).
+   1. Click **Enable publishing with Google Cloud Service account**.
+      The **Service account email** field displays.
+   1. Type the email of the service account into the
+      **Service account email** field.
+      You created this account in the previous step:
+      `pub-dev@$PROJECT_ID.iam.gserviceaccount.com`
+
+      ![Configuration that allows service account to publish on pub.dev](pub-dev-gcb-config.png)
```

Comment:
I think this is a separate step, ideally a separate section.

You might have a service account already.